File: FizzBuzz/BDD/JavaScript/src/FizzBuzz.js
class FizzBuzz {
    stringFor(input) {
	var result = this.fizzFilter(input)
	if (0 === input % 5) {
	    result += "Buzz"
	}
	if (result === '')
	{
	    return input.toString()
	}
	return result
    }

    fizzFilter(input) {
	if (0 === input % 3) {
	    return "Fizz"
	}
	return ''
    }
}

module.exports = { fizzBuzz: new FizzBuzz() }

File: FizzBuzz/BDD/RSpec/fizzbuzz.rb
class FizzBuzz
  attr_accessor :input

  def say(input)
    @input = input
  end

  def stringFor
    @result = filter(@input)
    @result.empty? ?
      @input.to_s
    :
      @result
  end

  def filter(input)
    @result = fizzFilter(input)
    @result.concat(buzzFilter(input))
    @result
  end

  def fizzFilter(integer)
    integer % 3 === 0 ?
      'Fizz'
    :
      ''
  end

  def buzzFilter(integer)
    integer % 5 === 0 ?
      'Buzz'
    :
      ''
  end
end
